src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -0,0 +1,293 @@
+            /// <list type="bullet">
+            @join param : format.resourceIdParams
+                /// <item><description>{@param.name}</description></item>
+            @end
+            /// </list>
+            /// </summary>
+            public static PathTemplate {@format.pathTemplateName} { get; } = new PathTemplate("{@format.pattern}");
+        
+            /// <summary>
+            /// Creates a {@format.entityName} resource name from its component IDs.
+            /// </summary>
+            @join param : format.resourceIdParams
+                /// <param name="{@param.name}Id">The {@param.name} ID.</param>
+            @end
+            /// <returns>
+            /// The full {@format.entityName} resource name.
+            /// </returns>
+            public static string {@format.name}({@formatResourceParams(format.resourceIdParams)}) => {@format.pathTemplateName}.Expand({@formatResourceArgs(format.resourceIdParams)});
+
+        @end
+        // Note: we could have parameterless overloads of Create and CreateAsync,
+        // documented to just use the default endpoint, settings and credentials.
+        // Pros:
+        // - Might be more reassuring on first use
+        // - Allows method group conversions
+        // Con: overloads!
+
+        @join method : api.apiMethods
+            {@xmlDoc(method.doc)}
+            {@method(method, FALSE)}
+
+        @end
+    }
+@end
+
+@private implClient(api)
+    public sealed partial class {@api.name}Impl : {@api.name}
+    {
+        private readonly ClientHelper _clientHelper;
+    }
+@end
+
+@private method(method, isImpl)
+    @switch method.type.toString
+    @case "FlattenedMethod"
+        {@flattenedMethodSync(method, isImpl)}
+    @case "FlattenedMethodAsyncCallSettings"
+        {@flattenedMethodAsyncCallSettings(method, isImpl)}
+    @case "FlattenedMethodAsyncCancellationToken"
+        {@flattenedMethodAsyncCancellationToken(method, isImpl)}
+    @default
+        *** ERROR: Cannot handle method type '{@method.type.toString}' ***
+    @end
+@end
+
+@private flattenedMethodSync(method, isImpl)
+    public virtual {@method.responseTypeName} {@method.name}(
+        {@objectParams(method.methodParamsWithExtras)})
+    {
+    @if isImpl
+    @else
+        {@""}    throw new NotImplementedException();
+    @end
+    }
+@end
+
+@private flattenedMethodAsyncCallSettings(method, isImpl)
+    public virtual {@method.responseTypeName} {@method.name}(
+        {@objectParams(method.methodParamsWithExtras)})
+    {
+    @if isImpl
+    @else
+        {@""}    throw new NotImplementedException();
+    @end
+    }
+@end
+
+@private flattenedMethodAsyncCancellationToken(method, isImpl)
+    @if isImpl
+    @else
+        public virtual {@method.responseTypeName} {@method.name}(
+            {@objectParams(method.methodParamsWithExtras)}) => {@method.name}(
+                {@objectArgs(method.methodParams)},
```

Comment:
I might prefer to use methodParams for the params showing up in the signature of our surface, and use a new attribute 'forwardingMethodParams' (or the like) for the params of the method we are forwarding the call to.